### ISSUE TITLE
Problem with Convert names, nodes and accessions to database

### ISSUE THREAD

**Issue by SergeyBaikal** [sherrillmix/taxonomizr#34]
Hello!

Using a linux system, I have the following error:
read.accession2taxid(list.files('.','prot.accession2taxid.FULL.gz$'),'accessionTaxa.sql')
Reading prot.accession2taxid.FULL.gz.
**Error in trimTaxa(ii, tmp, 1:3) : Malformed line on line 1** 

Can you please tell me how to solve these problems? 

if i used `Database('accessionTaxa.sql') ` starts downloading accession2taxid, but I have already downloaded the full version: prot.accession2taxid.FULL.gz.

**Comment by sherrillmix** [comment]
I hadn't heard of FULL taxaid before. Looks like they are a different format than standard taxaid files:

```
>zcat prot.accession2taxid.gz |head
accession	accession.version	taxid	gi
A0A009IHW8	A0A009IHW8.1	1310613	1835922267
A0A023FBW4	A0A023FBW4.1	34607	1939884164
A0A023FBW7	A0A023FBW7.1	34607	1939884197
```

```
>zcat prot.accession2taxid.FULL.gz |head
accession.version	taxid
0308206A	8058
0308221A	9606
0308230A	1049
```

so the read function is failing because the file is not in the expected format and it sees a "Malformed line on line 1".

The FULL format doesn't seem to be documented in the NCBI README:
https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/README
so I'm not totally sure what the point of it is although the name and file size imply it's some sort of more complete database (but what then is the normal non "FULL" version?). Why are you looking to use the FULL over prot.accession2taxid.gz?

**Comment by SergeyBaikal** [comment]
Thanks for the answer! I want to use the full version because I have tasks to identify the pool of virus sequences from different databases (RefSeq proteome, RefSeq genome). And since the virus databases are currently growing very quickly, the full version, I think, is a more reliable option.

**Comment by sherrillmix** [comment]
I think RefSeq is fairly curated subset of sequences so very likely to be included in the standard files. I'd guess you'd probably be served pretty well by the standard files. 

Otherwise, it wouldn't be that hard to parse the FULL but it would be a bit of a pain to actually use it because they don't include the versioned accession numbers so I would have to add special checks a to deal with a FULL vs normal-based database. 

So I guess my advice would be to try the standard accession2taxaid first and if that falls short or there's some sort of documentation why FULL is useful then we can move on from there.

**Comment by SergeyBaikal** [comment]
Thank you. I do not insist on redoing the file)) I was just curious to know what was going on. I found a way to convert IDs to names with FULL file. There is a Taxonkit program.

**Comment by sherrillmix** [comment]
Oh Taxonkit sounds nice. I hadn't seen that before. Wonder if it's new. Also do they do accession to taxa? I didn't see any mention of accession numbers and looks like they just talk about using taxdump which is just taxonomy so probably not. But if you're looking for just a few taxa or handy enough to convert accession to taxaid yourself then sounds like a good option.

**Comment by SergeyBaikal** [comment]
Yes, you are absolutely right. The program works with files from taxdmp files. I have a simple task. From IDs obtained in the Diamond program (-f 102), I need to get the full line of taxa.

**Comment by sherrillmix** [comment]
Oh well that's easier then. Also you technically didn't have to download any form of accession2taxaid at all then. They're only useful for converting accessions to taxaid. Good luck with the research.

**Comment by SergeyBaikal** [comment]
Thank you!
